addons/mail/models/mail_activity.py

```diff
@@ -127,9 +127,27 @@ def default_get(self, fields):
         'Document Name', compute='_compute_res_name', compute_sudo=True, store=True,
         help="Display name of the related document.", readonly=True)
     # activity
+    @api.model
+    def default_get_activity_type(self):
+        activity_type = self.env["mail.activity.type"]
+        ir_model_obj = self.env['ir.model']
+        current_model_name = self.env.context['default_res_model']
```

Comment:
Just to be complete, it would be better to call self.default_get(['res_model']). That way the ORM computes all possible ways to find a default res_model (using context or user-defined default values).